mypy_django_plugin/main.py

```diff
@@ -330,6 +334,13 @@ def get_attribute_hook(self, fullname: str) -> Optional[Callable[[AttributeConte
             return partial(request.set_auth_user_model_as_type_for_request_user, django_context=self.django_context)
         if info and info.has_base(fullnames.ABSTRACT_USER_MODEL_FULLNAME) and attr_name in ("is_staff", "is_active"):
             return partial(set_auth_user_model_boolean_fields, django_context=self.django_context)
+        if (
+            info
+            and info.has_base(fullnames.BASE_MANAGER_CLASS_FULLNAME)
+            and class_name in self._get_current_manager_bases()
+        ):
+            return partial(resolve_manager_method, django_context=self.django_context)
```

Comment:
Might be reasonable to add a comment here. Or lift out the checks in the statement into a descriptive variable name, so that it becomes visible what case we've found

e.g.

```suggestion
        is_dynamic_manager_class = (
            info
            and info.has_base(fullnames.BASE_MANAGER_CLASS_FULLNAME)
            and class_name in self._get_current_manager_bases()
        )
        if is_dynamic_manager_class:
            return partial(resolve_manager_method, django_context=self.django_context)
```